app/controllers/discovered_hosts_controller.rb

```diff
@@ -292,6 +293,21 @@ def find_by_name(*includes)
     @host
   end
 
+  def check_for_subnet
+    hosts_without_subnet = []
+    case params[:action]
+      when 'reboot', 'auto_provision'
+        if @host.subnet.nil?
+          process_warning :warning_msg => _("Discovered host reported from unknown subnet, communication will not be proxied.")
+        end
+      when 'submit_multiple_reboot', 'submit_multiple_auto_provision'
+        hosts_without_subnet = @hosts.limit(3).select { |host| host.subnet.nil? }.pluck(:name)
+        if hosts_without_subnet.present?
+          process_warning :warning_msg => _("Discovered hosts reported from unknown subnet are #{hosts_without_subnet}, communication will not be proxied.")
```

Comment:
Please do not use interpolation in gettext: https://projects.theforeman.org/projects/foreman/wiki/Translating#Translating-for-developers you need to use "%s" or hash for more keys.